java/modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/utils/sslcert/crl/CRLCache.java

```diff
@@ -114,7 +116,7 @@ private synchronized void replaceNewCacheValue(CRLCacheValue cacheValue) {
         }
 
         try {
-            String crlUrl = cacheValue.crlUrl;
+            @StartsWith({"https", "file"}) String crlUrl = cacheValue.crlUrl;
```

Comment:
Why do you need this annotation?

Reply:
It is sent to a method which calls new URL(crlUrl)

Reply:
Yes, but it should be inferred if no warning is issued when you write it. `crlUrl` is a local variable, so its type should be refined automatically.

Reply:
If adding this annotation caused the code to not issue a warning that was issued before, that is likely a bug in either our checker or the framework.